CI fails due to timing issue occasionally

Quite often one test fails due to timing issues making the whole CI pipeline fail, probably the time limit should be reduced slightly:
~~~
    def test_check_query_delay():
        with expected_protocol(ATSBase, [("TTIM?", "7")]) as inst:
            start = perf_counter()
            assert inst.maximum_test_time == 7
            delay = perf_counter() - start
>           assert delay > 0.05
E           assert 0.0491846000000038 > 0.05

tests\instruments\temptronic\test_temptronic_base.py:36: AssertionError
~~~

It did not work in [CI run](https://github.com/pymeasure/pymeasure/actions/runs/9091306319/job/24985650504?pr=1039):
~~~
    def test_check_query_delay():
        with expected_protocol(ATSBase, [("TTIM?", "7")]) as inst:
            start = perf_counter_ns()
            assert inst.maximum_test_time == 7
            delay = perf_counter_ns() - start
>           assert delay > 0.05 * 1e9
E           assert 47828800 > (0.05 * 1000000000.0)
~~~

--- a/tests/instruments/temptronic/test_temptronic_base.py
+++ b/tests/instruments/temptronic/test_temptronic_base.py
@@ -33,4 +33,7 @@ def test_check_query_delay():
         start = perf_counter_ns()
         assert inst.maximum_test_time == 7
         delay = perf_counter_ns() - start
-        assert delay > 0.05 * 1e9
+        # HACK acceptable factor is needed, as in CI under windows (Py38, Py39) the `sleep` interval
+        # is slightly shorter than the given argument.
+        acceptable_factor = 0.95
+        assert delay > 0.05 * 1e9 * acceptable_factor

